### ISSUE TITLE
Searching with email address no longer possible after migrating to SharePoint 2019

### ISSUE THREAD

**Issue by vinodkumarbmw** [Yvand/LDAPCP#154]
We have recently setup a new farm using SharePoint 2019 which replicates the existing SharePoint 2013 farm. We have configured LDAPCP as claims provider exactly same as it was in SharePoint 2013. Claims mapping we are using is UPN.

The issue is, in SharePoint 2013 we were able to search the user with email address but in SharePoint 2019, it does not work. If we use sAmAccountName, it returns the results.

Email Address:
![image](https://user-images.githubusercontent.com/94671718/142578028-4edd6839-278c-46c2-9aa3-5fad3fdd6bb3.png)

sAmAccountName:
![image](https://user-images.githubusercontent.com/94671718/142578053-28d5ddc9-7407-460a-883e-03c8f55ee660.png)

Claims Provider:
![image](https://user-images.githubusercontent.com/94671718/142578519-ce6bb284-8100-4d24-a772-5490558bec7a.png)

Claim Types Configuration:
![image](https://user-images.githubusercontent.com/94671718/142578721-0c9a05ae-9f76-4dab-a1e0-ee899085e651.png)

Can you please guide how do we enable searching using email address?

Thanks in advance!!

**Comment by Yvand** [comment]
@vinodkumarbmw go to central administration > security > LDAPCP Claim types configuration page.
At the bottom of the table you click New Item > "Add a mapping between a LDAP object and the main corresponding object type" and fill entries like this:

![image](https://user-images.githubusercontent.com/8788631/142651443-84371304-525e-40f0-bdf6-bfd769c3fef3.png)

It should work immediately after this.

**Comment by vinodkumarbmw** [comment]
@Yvand Thanks for the quick assistance. 
Actually, LDAPCP is configured for external users which are stored in AD differently because the attributes are replicated from LARS directory (internal database for partners/suppliers).

So, for external users, email field was empty in AD so the above mapping did not work.

However, I ran PowerShell command to find out which property contained the email address. There, I found it was "externalAttribute10" property that stores email address. 

Finally, I added the mapping using this attribute and the search started working immediately.
